x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/step_configure_package.tsx

```diff
@@ -15,86 +15,92 @@ import {
 } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n/react';
 
-import type {
-  PackageInfo,
-  RegistryStream,
-  NewPackagePolicy,
-  NewPackagePolicyInput,
-} from '../../../types';
+import type { PackageInfo, NewPackagePolicy, NewPackagePolicyInput } from '../../../types';
 import { Loading } from '../../../components';
+import { getStreamsForInputType, doesPackageHaveIntegrations } from '../../../services';
 
 import type { PackagePolicyValidationResults } from './services';
 import { PackagePolicyInputPanel } from './components';
 
-const findStreamsForInputType = (
-  inputType: string,
-  packageInfo: PackageInfo
-): Array<RegistryStream & { data_stream: { dataset: string } }> => {
-  const streams: Array<RegistryStream & { data_stream: { dataset: string } }> = [];
-
-  (packageInfo.data_streams || []).forEach((dataStream) => {
-    (dataStream.streams || []).forEach((stream) => {
-      if (stream.input === inputType) {
-        streams.push({
-          ...stream,
-          data_stream: {
-            dataset: dataStream.dataset,
-          },
-        });
-      }
-    });
-  });
-
-  return streams;
-};
-
 export const StepConfigurePackagePolicy: React.FunctionComponent<{
   packageInfo: PackageInfo;
+  showOnlyIntegration?: string;
   packagePolicy: NewPackagePolicy;
   updatePackagePolicy: (fields: Partial<NewPackagePolicy>) => void;
   validationResults: PackagePolicyValidationResults;
   submitAttempted: boolean;
-}> = ({ packageInfo, packagePolicy, updatePackagePolicy, validationResults, submitAttempted }) => {
+}> = ({
+  packageInfo,
+  showOnlyIntegration,
+  packagePolicy,
+  updatePackagePolicy,
+  validationResults,
+  submitAttempted,
+}) => {
+  const hasIntegrations = useMemo(() => doesPackageHaveIntegrations(packageInfo), [packageInfo]);
+  const packagePolicyTemplates = useMemo(
+    () =>
+      showOnlyIntegration
+        ? (packageInfo.policy_templates || []).filter(
+            (policyTemplate) => policyTemplate.name === showOnlyIntegration
+          ) || []
```

Comment:
nitpick but there is probably a useless `|| []` here (if it can simplify that expression ) 
```suggestion
        ? (packageInfo.policy_templates || []).filter(
            (policyTemplate) => policyTemplate.name === showOnlyIntegration
          ) 
```